src/instance.rs

```diff
@@ -954,6 +961,35 @@ impl<'a, 'py, T> Borrowed<'a, 'py, T> {
     {
         FromPyObject::extract(self.to_any())
     }
+
+    /// Cast this to a concrete Python type or pyclass.
+    ///
+    /// This performs a runtime type check using the equivalent of Python's
+    /// `isinstance(self, U)`.
+    #[inline]
+    pub fn cast<U>(self) -> Result<Borrowed<'a, 'py, U>, DowncastError<'a, 'py>>
+    where
+        U: PyTypeCheck,
+    {
+        if U::type_check(&self.to_any()) {
```

Comment:
The implementations on `Bound` have a generic `inner` to narrow the generic parameters just to `U`, perhaps worth doing here?